velox/exec/tests/WindowTest.cpp

```diff
@@ -679,5 +679,63 @@ DEBUG_ONLY_TEST_F(WindowTest, reserveMemorySort) {
   }
 }
 
+TEST_F(WindowTest, NaNFrameBound) {
+  const auto kNan = std::numeric_limits<double>::quiet_NaN();
+  auto data = makeRowVector(
+      {"c0", "s0", "off0", "off1"},
+      {
+          makeFlatVector<int64_t>({1, 2, 3}),
+          makeFlatVector<double>({1.0, 2.0, 3.0}),
+          makeFlatVector<double>({0.1, 2.0, 1.9}),
+          makeFlatVector<double>({kNan, 2.0, kNan}),
+      });
+
+  const auto makeFrames = [](const std::string& call) {
+    std::vector<std::string> frames;
+
+    std::vector<std::string> orders{"asc", "desc"};
+    std::vector<std::string> bounds{"preceding", "following"};
+    for (const std::string& order : orders) {
+      for (const std::string& startBound : bounds) {
+        for (const std::string& endBound : bounds) {
+          // Frames starting from following and ending at preceding are not
+          // allowed.
+          if (startBound == "following" && endBound == "preceding") {
+            continue;
+          }
+          frames.push_back(fmt::format(
+              "{} over (order by s0 {} range between off0 {} and off1 {})",
+              call,
+              order,
+              startBound,
+              endBound));
+          frames.push_back(fmt::format(
+              "{} over (order by s0 {} range between off1 {} and off0 {})",
+              call,
+              order,
+              startBound,
+              endBound));
+        }
+      }
+    }
+    return frames;
+  };
+
+  auto expected = makeRowVector(
+      {makeNullableFlatVector<int64_t>({std::nullopt, 2, std::nullopt})});
+  for (const auto& frame : makeFrames("sum(c0)")) {
+    auto plan =
+        PlanBuilder().values({data}).window({frame}).project({"w0"}).planNode();
+    AssertQueryBuilder(plan).assertResults(expected);
+  }
+
+  expected = makeRowVector({makeFlatVector<int64_t>({1, 2, 3})});
+  for (const auto& frame : makeFrames("rank()")) {
```

Comment:
@kagamiori : rank() is not affected by frame values. Only value functions (first_value, last_value, nth_value) and aggregate functions are affected by frame boundaries. MIght be better to use one of those functions in the test.

Reply:
Hi @aditi-pandit, I intentionally added the test case of rank() to ensure it is not affected by the handling of NaN frame bound as you point out. The regular case is tested by the sum() function above.

Reply:
@kagamiori : I see. Maybe then add a comment to indicate this is for a function not affected by the Nan frames.

Reply:
@aditi-pandit, sure, will do. Thanks for the suggestion!